src/game/scripts/vscripts/abilities/pudge_flesh_heap_health_regeneration.lua

```diff
@@ -0,0 +1,143 @@
+
+function modifier_flesh_heap_health_regeneration:IsPassive()
+    return true
+end
+
+function modifier_flesh_heap_health_regeneration:IsPurgable()
+    return true
+end
+
+function modifier_flesh_heap_health_regeneration:GetFleshHeapKills()
+  if self.nKills == nil then
+    self.nKills = 0
+  end
+  return self.nKills
+end
+ 
+--------------------------------------------------------------------------------
+
+function modifier_flesh_heap_health_regeneration:OnCreated( kv )
+  if not self:GetAbility() then
+    self:GetParent():RemoveModifierByName("modifier_flesh_heap_health_regeneration")
+    self:GetParent():CalculateStatBonus()
+    return
+  end
+  self.flesh_heap_value_buff_amount = self:GetAbility():GetSpecialValueFor( "flesh_heap_value_buff_amount" ) or 0
+  if IsServer() then
+    self:SetStackCount( self:GetFleshHeapKills() )
+    self:GetParent():CalculateStatBonus()
+  end
+end
+
+--------------------------------------------------------------------------------
+
+function modifier_flesh_heap_health_regeneration:OnRefresh( kv )
+  if not self:GetAbility() then
+    self:GetParent():RemoveModifierByName("modifier_flesh_heap_health_regeneration")
+    self:GetParent():CalculateStatBonus()
+    return
+  end
+  self.flesh_heap_value_buff_amount = self:GetAbility():GetSpecialValueFor( "flesh_heap_value_buff_amount" ) or 0
+  if IsServer() then
+    self:SetStackCount( self:GetFleshHeapKills() )
+    self:GetParent():CalculateStatBonus()
+  end
+end
+
+--------------------------------------------------------------------------------
+
+function modifier_flesh_heap_health_regeneration:DeclareFunctions()
+  local funcs = {
+    MODIFIER_PROPERTY_HEALTH_REGEN_PERCENTAGE,
+    MODIFIER_EVENT_ON_DEATH,
+  }
+  return funcs
+end
+
+function modifier_flesh_heap_health_regeneration:OnDeath(keys)
+
+
+  if not keys.unit or not keys.attacker then 
+    return 
+  end
+
+  if not keys.unit:IsRealHero() then
+    return 
+  end
+
+  if keys.unit:IsTempestDouble() then
+    return
+  end
+
+  if not IsServer() then 
+    return 
+  end
+  -----------------------------------------------------------------------------
+  local hKiller = keys.attacker
+  local hVictim = keys.unit
+
+
+  if self:GetCaster():GetTeamNumber() ~= hVictim:GetTeamNumber() then
+    self.fleshHeapRange = self:GetAbility():GetSpecialValueFor( "flesh_heap_range")
+    local vToCaster = self:GetCaster():GetOrigin() - hVictim:GetOrigin()
+    local flDistance = vToCaster:Length2D() - (self:GetCaster():GetCollisionPadding() + hVictim:GetCollisionPadding())
+    if hKiller == self:GetCaster() or self.fleshHeapRange >= flDistance then
+      if self.nKills == nil then
+        self.nKills = 0
+      end
+
+      self.nKills = self.nKills + 1
+
+      local hBuff = self:GetCaster():FindModifierByName( "modifier_flesh_heap_health_regeneration" )
+      if hBuff ~= nil then
+        hBuff:SetStackCount( self.nKills )
```

Comment:
Is it really needed to check if this modifier is present? Intrinsic modifiers are managed automatically by the game, and as long as you still have the ability it should be present at all times. I'm not sure it's necessary.